File: scripts/check-product-colors.js
// Script to check which products are missing color options
import { getAllProducts } from '../app/lib/shopify.js';

async function checkProductColors() {
  console.log('Checking all products for color options...');
  
  try {
    // Get all products
    const products = await getAllProducts();
    console.log(`Found ${products.length} products total.`);
    
    const productsWithoutColors = [];
    const productsWithColors = [];
    
    // Check each product for color options
    for (const product of products) {
      let hasColorOption = false;
      
      // Check in product options
      if (product.options) {
        const colorOption = product.options.find(option => 
          option.name.toLowerCase() === 'color'
        );
        
        if (colorOption && colorOption.values.length > 0) {
          hasColorOption = true;
          productsWithColors.push({
            title: product.title,
            handle: product.handle,
            colors: colorOption.values
          });
          continue;
        }
      }
      
      // Check in variants
      if (product.variants && product.variants.edges.length > 0) {
        const colorSet = new Set();
        
        for (const { node: variant } of product.variants.edges) {
          if (variant.selectedOptions) {
            const colorOpt = variant.selectedOptions.find(opt => 
              opt.name.toLowerCase() === 'color'
            );
            
            if (colorOpt) {
              colorSet.add(colorOpt.value);
            }
          }
        }
        
        if (colorSet.size > 0) {
          hasColorOption = true;
          productsWithColors.push({
            title: product.title,
            handle: product.handle,
            colors: Array.from(colorSet)
          });
          continue;
        }
      }
      
      // If no color options found
      if (!hasColorOption) {
        productsWithoutColors.push({
          title: product.title,
          handle: product.handle,
          productType: product.productType,
          tags: product.tags
        });
      }
    }
    
    // Display results
    console.log('\n=== Products WITH Color Options ===');
    console.log(`Total: ${productsWithColors.length}`);
    productsWithColors.forEach(product => {
      console.log(`- ${product.title} (${product.handle})`);
      console.log(`  Colors: ${product.colors.join(', ')}`);
    });
    
    console.log('\n=== Products WITHOUT Color Options ===');
    console.log(`Total: ${productsWithoutColors.length}`);
    productsWithoutColors.forEach(product => {
      console.log(`- ${product.title} (${product.handle})`);
      console.log(`  Type: "${product.productType}", Tags: ${product.tags.join(', ')}`);
    });
    
    console.log('\nSummary:');
    console.log(`Total products: ${products.length}`);
    console.log(`Products with color options: ${productsWithColors.length}`);
    console.log(`Products without color options: ${productsWithoutColors.length}`);
    
  } catch (error) {
    console.error('Error checking product colors:', error);
  }
}

// Run the function
checkProductColors()
  .catch(error => {
    console.error('Error in script execution:', error);
  });

File: scripts/check-product-colors.py
#!/usr/bin/env python3
import os
import requests
import json
from dotenv import load_dotenv

# Load environment variables from .env.local file
load_dotenv(dotenv_path=os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), '.env.local'))

def check_product_colors():
    # Get Shopify credentials from environment variables
    store_domain = os.getenv('SHOPIFY_STORE_DOMAIN')
    token = os.getenv('SHOPIFY_STOREFRONT_API_TOKEN')
    
    if not store_domain or not token:
        print("Error: Missing Shopify API credentials. Please check your .env file.")
        return
    
    # Shopify Storefront API endpoint
    endpoint = f"https://{store_domain}/api/2023-07/graphql.json"
    
    # Headers for API request
    headers = {
        'X-Shopify-Storefront-Access-Token': token,
        'Content-Type': 'application/json',
        'Accept': 'application/json'
    }
    
    # GraphQL query to get all products with their options and variants
    query = """
    {
      products(first: 100) {
        edges {
          node {
            id
            title
            handle
            productType
            tags
            options {
              name
              values
            }
            variants(first: 20) {
              edges {
                node {
                  id
                  title
                  selectedOptions {
                    name
                    value
                  }
                }
              }
            }
          }
        }
      }
    }
    """
    
    # Make the API request
    try:
        response = requests.post(
            endpoint,
            headers=headers,
            json={"query": query}
        )
        
        response.raise_for_status()  # Raise an exception for HTTP errors
        data = response.json()
        
        # Process the products
        products = data.get('data', {}).get('products', {}).get('edges', [])
        products_with_colors = []
        products_without_colors = []
        
        print(f"Found {len(products)} products total.")
        
        for product_edge in products:
            product = product_edge['node']
            has_color_option = False
            
            # Check in product options
            color_values = []
            for option in product.get('options', []):
                if option['name'].lower() == 'color':
                    has_color_option = True
                    color_values = option['values']
                    products_with_colors.append({
                        'title': product['title'],
                        'handle': product['handle'],
                        'colors': color_values
                    })
                    break
            
            if has_color_option:
                continue
            
            # Check in variants
            color_set = set()
            for variant_edge in product.get('variants', {}).get('edges', []):
                variant = variant_edge['node']
                for option in variant.get('selectedOptions', []):
                    if option['name'].lower() == 'color':
                        color_set.add(option['value'])
            
            if color_set:
                has_color_option = True
                products_with_colors.append({
                    'title': product['title'],
                    'handle': product['handle'],
                    'colors': list(color_set)
                })
                continue
            
            # If no color options found
            if not has_color_option:
                products_without_colors.append({
                    'title': product['title'],
                    'handle': product['handle'],
                    'productType': product['productType'],
                    'tags': product['tags']
                })
        
        # Display results
        print("\n=== Products WITH Color Options ===")
        print(f"Total: {len(products_with_colors)}")
        for product in products_with_colors:
            print(f"- {product['title']} ({product['handle']})")
            print(f"  Colors: {', '.join(product['colors'])}")
        
        print("\n=== Products WITHOUT Color Options ===")
        print(f"Total: {len(products_without_colors)}")
        for product in products_without_colors:
            print(f"- {product['title']} ({product['handle']})")
            print(f"  Type: \"{product['productType']}\", Tags: {', '.join(product['tags'])}")
        
        print("\nSummary:")
        print(f"Total products: {len(products)}")
        print(f"Products with color options: {len(products_with_colors)}")
        print(f"Products without color options: {len(products_without_colors)}")
        
    except requests.exceptions.RequestException as e:
        print(f"Error making API request: {e}")
    except Exception as e:
        print(f"Unexpected error: {e}")

if __name__ == "__main__":
    check_product_colors()
